packages/zoe/src/zoeService/invitationQueries.js

```diff
@@ -4,36 +4,33 @@ import { assert, details as X } from '@agoric/assert';
 import { E } from '@agoric/eventual-send';
 
 export const makeInvitationQueryFns = invitationIssuer => {
-  /** @type {GetAmountOfInvitationThen} */
-  const getAmountOfInvitationThen = async (invitationP, onFulfilled) => {
-    const onRejected = () =>
-      assert.fail(X`A Zoe invitation is required, not ${invitationP}`);
-    return E(invitationIssuer)
-      .getAmountOf(invitationP)
-      .then(onFulfilled, onRejected);
+  /** @type {GetInvitationDetails} */
+  const getInvitationDetails = async invitationP => {
+    const onRejected = reason => {
+      // Or you could do it the other way around
+      assert.note(reason, X`A Zoe invitation is required, not ${invitationP}`);
+      throw reason;
+    };
```

Comment:
The other way around should keep the error you want at the top, while still linking to the original error
```suggestion
    const onRejected = reason => {
      const err = assert.error(X`A Zoe invitation is required, not ${invitationP}`);
      assert.note(err, X`Due to ${reason}`);
      throw err;
    };
```